config.cpp

```diff
@@ -323,57 +323,71 @@ static int parse_channels(libconfig::Setting &chans, device_t *dev, int i) {
 				for(int f = 0; f<channel->freq_count; f++) {
-					channel->freqlist[f].squelch.set_squelch_value((int)chans[j]["squelch"][f]);
+					float threshold_dBFS = (float)chans[j]["squelch_threshold"][f];
+					if (threshold_dBFS > 0) {
+						cerr << "Configuration error: devices.["<<i<<"] channels.["<<j<<"]: squelch_threshold must be less than or equal to 0\n";
+						error();
+					} else if (threshold_dBFS == 0) {
+						channel->freqlist[f].squelch.set_squelch_level_threshold(0);
+					} else {
+						channel->freqlist[f].squelch.set_squelch_level_threshold(dBFS_to_level(threshold_dBFS));
+					}
 				}
-				// NB: no value check; -1 allows auto-squelch for
-				//     some frequencies and not others.
-			} else if(libconfig::Setting::TypeInt == chans[j]["squelch"].getType()) {
+			} else if(libconfig::Setting::TypeInt == chans[j]["squelch_threshold"].getType()) {
 				// Legacy (single squelch for all frequencies)
-				int sqlevel = (int)chans[j]["squelch"];
-				if(sqlevel < 0) {
-					cerr<<"Configuration error: devices.["<<i<<"] channels.["<<j<<"]: squelch must be greater than 0\n";
+				float threshold_dBFS = (float)chans[j]["squelch_threshold"];
+				float level;
+				if (threshold_dBFS > 0) {
+					cerr << "Configuration error: devices.["<<i<<"] channels.["<<j<<"]: squelch_threshold must be less than or equal to 0\n";
 					error();
+				} else if (threshold_dBFS == 0) {
+					level = 0;
+				} else {
+					level = dBFS_to_level(threshold_dBFS);
 				}
 				for(int f = 0; f<channel->freq_count; f++) {
-					channel->freqlist[f].squelch.set_squelch_value(sqlevel);
+					channel->freqlist[f].squelch.set_squelch_level_threshold(level);
 				}
 			} else {
-				cerr<<"Invalid value for squelch (should be int or list - use parentheses)\n";
+				cerr << "Invalid value for squelch_threshold (should be int or list - use parentheses)\n";
 				error();
 			}
 		}
-		if(chans[j].exists("squelch_db")) {
-			if(libconfig::Setting::TypeList == chans[j]["squelch_db"].getType()) {
+		if(chans[j].exists("squelch_snr_threshold")) {
+			// Value is SNR in dB, zero disables squelch (ie always open), positive is valid, negative is invalid
+			if(libconfig::Setting::TypeList == chans[j]["squelch_snr_threshold"].getType()) {
 				// New-style array of per-frequency squelch settings
 				for(int f = 0; f<channel->freq_count; f++) {
-					channel->freqlist[f].squelch.set_squelch_db((float)chans[j]["squelch_db"][f]);
+					float snr = (float)chans[j]["squelch_snr_threshold"][f];
+					if (snr < 0) {
+						cerr << "Configuration error: devices.["<<i<<"] channels.["<<j<<"]: squelch_snr_threshold must be greater than or equal to 0\n";
+						error();
+					} else {
+						channel->freqlist[f].squelch.set_squelch_snr_threshold(snr);
+					}
 				}
-			} else if(libconfig::Setting::TypeFloat == chans[j]["squelch_db"].getType()) {
+			} else if(libconfig::Setting::TypeFloat == chans[j]["squelch_snr_threshold"].getType()) {
```

Comment:
`squelch_threshold` is `TypeInt`, but `squelch_snr_threshold` is `TypeFloat` - that's confusing.

Reply:
That is also an option, however one will have to use floats even when not intending to use fractional decibels. That's a bit inconvenient. Also specifying squelch thresholds with a sub-decibel resolution is a bit of an overkill, no?

Reply:
> Also specifying squelch thresholds with a sub-decibel resolution is a bit of an overkill, no?

Yeah, maybe a bit of overkill ;)

For `squelch_threshold` given that is dBFS and the waterfall is only showing whole numbers, `int` make sense, I'll change that.

For `squelch_snr_threshold` I'm planning to use fractional dB so I can get more control (`9` gives a multiplier of `2.81` and `10` gives `3.16`).  I'll make the config support either `int` or `float` so it's easier for those who don't want to use decimal values.